### ISSUE TITLE
Implementing Gradient Accumulation in SentenceBERT

### ISSUE THREAD

**Issue by sancharijoveo** [UKPLab/sentence-transformers#1570]
Hi, can anyone help me understand where I can include the gradient accumulation argument in the fine-tuning of a roberta-base model using sentence-transformers? I have tried including gradient_accumulation_steps directly as a parameter in model.fit() as well as a part of optimizer_params dict in model.fit(), it is not accepted as a valid parameter in both cases. 

Thanks in advance :)

**Comment by sancharijoveo** [comment]
Thanks @nreimers for your quick response. I wanted to implement gradient accumulation to get over the limitation of my GPU memory. Can you direct me to any resource/code which I can look at to implement it ad-hoc during fine-tuning?

**Comment by nreimers** [comment]
What is the current batch size you can run?

**Comment by sancharijoveo** [comment]
I can run 16 for roberta-base, 32 for distilroberta-base and only 2 for longformer-base-4096 (with max_seq_length = 786). The GPU memory is 16GB.

**Comment by nreimers** [comment]
For using gradient_accumulation_steps, the training loop must be changed.

But adding gradient_accumulation_steps will not improve your model. You will end up with the quality of model independent if you train with or without gradient_accumulation_steps.

**Comment by sancharijoveo** [comment]
The training loop code is the fit() function in https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/SentenceTransformer.py?

Also, wouldn't gradient accumulation help mimic a larger batch size when there's a limitation on the GPU memory?
